mysql-test/suite/galera_encryption/t/gcache_mk_rotation.test

```diff
@@ -0,0 +1,90 @@
+#
+# Test rotation of GCache Master Key
+#
+
+--source include/galera_cluster.inc
+
+--connection node_1
+--let $NODE1_DATA_DIR = `select @@datadir`
+--connection node_2
+--let $NODE2_DATA_DIR = `select @@datadir`
+
+--connection node_1
+
+# In this MTR suite the server is started with GCache encryption enabled
+CREATE TABLE t1(a INT PRIMARY KEY);
+
+# Check that GCache ring buffer file content is encrypted
+--let $assert_text = Assert that GCache is encrypted
+--let $assert_file = $NODE1_DATA_DIR/galera.cache
+--let $assert_select = enc_encrypted: 1
+--let $assert_count = 1
+--source include/assert_grep.inc
+
+# Check that GCache ring buffer file is using MK id 1
+--let $assert_text = Assert that GCache uses MK id 1
+--let $assert_file = $NODE1_DATA_DIR/galera.cache
+--let $assert_select = enc_mk_id: 1
+--let $assert_count = 1
+--source include/assert_grep.inc
+
+--connection node_2
+
+# Check that GCache ring buffer file content is encrypted
+--let $assert_text = Assert that GCache is encrypted
+--let $assert_file = $NODE2_DATA_DIR/galera.cache
+--let $assert_select = enc_encrypted: 1
+--let $assert_count = 1
+--source include/assert_grep.inc
+
+# Check that GCache ring buffer file is using MK id 1
+--let $assert_text = Assert that GCache uses MK id 1
+--let $assert_file = $NODE2_DATA_DIR/galera.cache
+--let $assert_select = enc_mk_id: 1
+--let $assert_count = 1
+--source include/assert_grep.inc
+
+ALTER INSTANCE ROTATE GCACHE MASTER KEY;
```

Comment:
What if alter instance ... is executed without wsrep_on, should that work, and rotate the key only on one server?

Is it possible that one node uses gcache encryption, the other doesn't?

Reply:
This is not wsre_on -aware. It works the same way as ALTER INSTANCE ROTATE INNODB MASTER KEY works.
I will add test scenario for this.

Reply:
MTR test added